app/src/main/java/com/kickstarter/libs/LakeTrackingClient.kt

```diff
@@ -0,0 +1,80 @@
+package com.kickstarter.libs
+
+import android.content.Context
+import android.os.Bundle
+import android.util.Log
+import com.kickstarter.libs.qualifiers.ApplicationContext
+import com.kickstarter.libs.utils.MapUtils
+import com.kickstarter.models.User
+import com.kickstarter.services.LakeBackgroundService
+import com.kickstarter.services.firebase.dispatchJob
+import com.kickstarter.ui.IntentKey
+import io.fabric.sdk.android.Fabric
+import org.json.JSONException
+import org.json.JSONObject
+import timber.log.Timber
+import java.util.*
+
+class LakeTrackingClient(
+        @param:ApplicationContext private val context: Context,
+        currentUser: CurrentUserType,
+        build: Build,
+        currentConfig: CurrentConfigType) : TrackingClient(context, currentUser, build, currentConfig) {
+    private var loggedInUser: User? = null
+    private var config: Config? = null
+
+    init {
+
+        // Cache the most recent logged in user for default Lake properties.
+        this.currentUser.observable().subscribe { u -> this.loggedInUser = u }
+
+        // Cache the most recent config for default Lake properties.
+        this.currentConfig.observable().subscribe { c -> this.config = c }
+    }
+
+    override fun track(eventName: String, additionalProperties: Map<String, Any>) {
```

Comment:
This function is very similar to the one existing on the `KoalaTrackingClient`. I wonder if we could move it to the parent class and call it from the children passing the parameters we need (`uniqueJobName`, `eventName`, `data`, etc)

Reply:
They depend on different services `LakeBackgroundService` vs `KoalaBackgroundService` and the 2 services depend on different `IntentKey`s. So would I also be passing in the service class's name,`TAG`, and `IntentKey`s?

Reply:
Yes... whatever differs from each other we would pass as argument. What do you think?

Reply:
I can try it and open it as a PR against this branch since it'll touch 3 files!